source/jsonifiable.d.ts

```diff
@@ -0,0 +1,32 @@
+import type {JsonPrimitive, JsonValue} from './basic';
+
+type JsonifiableObject = {[Key in string]?: Jsonifiable} | {toJSON: () => JsonValue};
```

Comment:
Shouldn't this look deeply for `toJSON`?

See https://github.com/sindresorhus/type-fest/commit/e11d30b727fc8b3e9438fbf15ec8619449e5566e and https://github.com/sindresorhus/type-fest/issues/468

Reply:
Could you give me a test case that you're worried about? I'm not sure I understand your concern. I added this test case as a guess of what you mean though:

```typescript
expectAssignable<Jsonifiable>({a: {deeply: {nested: {toJsonObject: new Date()}}}});
```

Reply:
Disregard my last comment. I must have messed up the test I did which prompted https://github.com/sindresorhus/type-fest/pull/492/commits/d8d78f8c052561b8fefbe431c95cb6f507e2b8e1. I'll roll back that change and add a test.